### DISCUSSION TITLE
Embed wiki plugin markup in template/iterator

### DISCUSSION THREAD

**Discussion by listerr** [combostrap/combo#38] [General]
Hi,

Taking your example of a content-list iterator:

```
<content-list flush color="info" width="500px">

<iterator>
    <data>select from pages where is_low_quality = 0 and path glob ':reference:*'</data>
<template>    
 <row>
        <itext><icon name="file"/> [[$path?preview|$name]] </itext>
         <badge align="right">{{tagpage>$keywords}}</badge>
 </row>
</template>
 
</iterator>
</content-list>
```

I was trying to make the badge more useful, clickable link to the tag plugin to get a list of all tags, i,e,
to use the keywords to make a link to tags.

It just outputs a badge with the bare "$keywords"

I can hard link it with a URL:

`<badge align="right">[[https://my,site.here/tag?do=showtag&tag=$keywords|$keywords]]</badge>`

Wondered if there was a better method of adding a label for keywords / tags in a content list?

Thanks

**Comment by gerardnico** [comment]
We don't have `keyword` as a [variable](https://combostrap.com/templating/all-variables-that-can-be-used-in-a-template-text-qfr74w67)
I have added it see #39 

You could then be able to process it with a [pipeline](https://combostrap.com/templating/pipeline-rx8speed).

In the next release, this is also possible to show the ascendants or the parent  (with link) via a enhanced [hierarchical breadcrumb](https://combostrap.com/navigation/combostrap-ui-hierarchical-breadcrumb-q6kfagq5)

**Comment by gerardnico** [comment]
Note that you can create a relative link.

```html
<badge align="right">[[?do=showtag&tag=$keywords|$keywords]]</badge>
```

**Reply by listerr** [reply]
I tried that, but it doesn't seem to work when I do relative URL because DW escapes it. 

The `&` becomes `&amp;` breaking the URL:

https://my.site.com/reference?do=showtag&amp;tag=law

**Comment by gerardnico** [comment]
The addition is for the next release.

I have also update the branch [1.24.1](https://github.com/ComboStrap/combo/archive/refs/heads/1.24.1.zip).
Or you can do change also manually: See 3afffe6a716ac464c5b0311681966da633c57602

**Comment by gerardnico** [comment]
Our bad.

We had tested with the path ie
```
[[path?do=...|label]]
```
but not
```
[[?do=...|label]]
```
It's corrected in the [patch branch](https://github.com/ComboStrap/combo/archive/refs/heads/1.24.1.zip)
via this commit 4e1ec12dbb63e1b0ce9383d42f9d555ef34ba983

If you install it manually, it should work

**Reply by gerardnico** [reply]
And I see that i messed up with the comment hierarchy ... sorry

**Comment by listerr** [comment]
I am probably trying to use this in the wrong way, as I use the (very useful!) tag + pagelist plugins to generate nice
lists of pages based on tags etc. (i.e. as there is no way within the iterator to do "get a list of tags for this page") 
I was abusing the keywords metadata.

https://www.dokuwiki.org/plugin:tag 
https://www.dokuwiki.org/plugin:pagelist#flags

It can generate lists of pages in a particular namespace based on tags e.g.

`{{topic>systems?mainpage&simplelist}}`

Had a look at the code to see how to get page tags - 

It seems quite easy to obtain the page tags, as it uses an existing [dokuwiki metadata](https://www.dokuwiki.org/devel:metadata) field called 'subject'.

```php
<php>
$id = cleanID(getID());
$tags = p_get_metadata($id, 'subject');
$taglist = implode(', ', $tags);
echo "Tags: $taglist\n";
</php>

```
I had a look at the ComboStrap code to see if I could add this, but there are a lot of pending changes between 1.24 and 1.25,
so I didn't try to hack it.

This might be a useful field to add to the iterator/metadata/PageSQL anyway, as tags are used by a few other plugins.

Rob

**Reply by gerardnico** [reply]
As of today, a tag is possible with backlinks.

Unfortunately, there is not yet a good solution to show them in a iterator.

**Reply by gerardnico** [reply]
The long term view on this is that a tag is a link to a page that comes extend the hierarchical breadcrumb.
 
Not yet done unfortunately.
